tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -3536,6 +3545,12 @@ SpirvVariable *DeclResultIdMapper::getBuiltinVar(spv::BuiltIn builtIn,
   return var;
 }
 
+SpirvVariable *DeclResultIdMapper::getBuiltinVar(spv::BuiltIn builtIn,
+                                                 QualType type,
+                                                 SourceLocation loc) {
+  return getBuiltinVar(builtIn, type, spv::StorageClass::Max, loc);
```

Comment:
I would move the code that infers the storage class based on `builtIn` into this function, and pass in that storage class instead. There is no need to pass in `Max`.

Reply:
That's fair, I think I was trying to avoid needing to pass in `needsLegalization` and `mayNeedFlatDecoration` as well, but looks like I'll need to look closer at `mayNeedFlatDecoration` too.